src/Workspaces/Core/Portable/Utilities/Operations/AbstractOperationContext.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Utilities
+{
+    /// <summary>
+    /// Abstract base implementation of the <see cref="IOperationContext"/> interface.
+    /// </summary>
```

Comment:
this code was effectively copied near verbatim (with just mechanical changes) from https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=%2Fsrc%2FEditor%2FText%2FDef%2FTextUI%2FUtilities%2FAbstractUIThreadOperationContext.cs&_a=contents&version=GBmain